### DISCUSSION TITLE
Training on dataset with different sets of data

### DISCUSSION THREAD

**Discussion by alikhamze** [deepmodeling/deepmd-kit#2609] [Q&A] [answered]
Hello,

I have a dataset where one portion includes virial data and the other does not.

If I train deepmd using the full dataset and want it to predict the virial as well, how are the data points without virial data handled? Will having the mixed dataset affect my virial predictions (beyond having fewer virial data points than force and energy data points)?

Thanks,
-Ali

**Comment by njzjz** [comment, marked as answer]
If a data set doesn't contain virials, the total loss for this data will not have the virial part.

We note that there is a known bug that virial loss will still exist in `lcurve.out` even if the total loss doesn't contain the virial loss.

**Reply by alikhamze** [reply]
Hi @njzjz, thank you for the reply.

So it sounds like training should proceed without an issue with the mixed dataset, and that the due to the bug the rmse_v_train column of the lcurve.out will always be printed, even when the data points in the batch don't have virial data?

**Reply by njzjz** [reply]
Yes

**Reply by alikhamze** [reply]
great, thank you!
